app/controllers/AmendablePOAController.scala

```
@@ -0,0 +1,85 @@
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class AmendablePOAController @Inject()(val authorisedFunctions: AuthorisedFunctions,
+                                       claimToAdjustService: ClaimToAdjustService,
+                                       val auth: AuthenticatorPredicate,
+                                       view: AmendablePaymentOnAccount,
+                                       implicit val itvcErrorHandler: ItvcErrorHandler,
+                                       implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler)
+                                      (implicit val appConfig: FrontendAppConfig,
+                                       implicit override val mcc: MessagesControllerComponents,
+                                       val ec: ExecutionContext)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching with ImplicitCurrencyFormatter {
+
+  def show(isAgent: Boolean): Action[AnyContent] =
+    auth.authenticatedAction(isAgent) {
+      implicit user =>
+        if (isEnabled(AdjustPaymentsOnAccount)) {
+          claimToAdjustService.getPoaForNonCrystallisedTaxYear(Nino(user.nino)) flatMap {
+            case Right(Some(poa: PaymentOnAccount)) =>
+              Future.successful(
+                Ok(view(
+                  isAgent = isAgent,
+                  taxYearModel = poa.taxYear,
+                  poaOneTransactionId = poa.poaOneTransactionId,
+                  poaTwoTransactionId = poa.poaTwoTransactionId,
+                  poaOneFullAmount = poa.paymentOnAccountOne,
+                  poaTwoFullAmount = poa.paymentOnAccountTwo
+                ))
+              )
+            case Right(None) =>
+              Logger("application").error(s"Failed to create PaymentOnAccount model")
+              Future.successful(showInternalServerError(isAgent))
+            case Left(ex) =>
+              Logger("application").error(s"Exception: ${ex.getMessage} - ${ex.getCause}")
```

I guess you can just throw Exception here instead of logging and failing future